### PR TITLE
feat: add keepAlive method in Stream.

### PR DESCRIPTION
**Pull request by fmeriaux**
Hello, 

Here is a modest contribution to add a keepAlive on streams, very useful when handling TCP connections with proxies that have timeouts.

### CODE REVIEW

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -760,6 +760,24 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
   def delayBy[F2[x] >: F[x]: Temporal](d: FiniteDuration): Stream[F2, O] =
     Stream.sleep_[F2](d) ++ this
 
+  /** Ensure that the stream always emits elements by defining a maxIdle duration.
+    * In other words, the stream will emit an element when it hasn't emitted any since the maximum time specified.
+    */
+  def keepAlive[F2[x] >: F[x]: Temporal, O2 >: O](
+      maxIdle: FiniteDuration,
+      heartbeat: O2
```
Comment (He-Pin) [inline core/shared/src/main/scala/fs2/Stream.scala:768]: Better to be a function.

Comment (fmeriaux) [reply core/shared/src/main/scala/fs2/Stream.scala:768]: You suggest, update to : `heartbeat: => O2` ?

**Comment by fmeriaux** [comment]
Do you have any idea why I have undefined entries in my test? 

```
=> Obtained
List(
  1,
  2,
  0,
  undefined,
  3,
  4,
  0,
  0,
  undefined,
  5
)
=> Diff (- obtained, + expected)
   0,
-  undefined,
   3,
   0,
-  undefined,
   5
```

File: core/shared/src/main/scala/fs2/Stream.scala (hunk 1 above)
Comment (He-Pin) [reply core/shared/src/main/scala/fs2/Stream.scala:768]: Yes, a user may provide different heart beat messages, iirc, pekko akka stream is act like that.

Comment (fmeriaux) [reply core/shared/src/main/scala/fs2/Stream.scala]: I thought about it but couldn't figure out how to make a Pull.output from a F effect.

Comment (mpilquist) [reply core/shared/src/main/scala/fs2/Stream.scala]: `Pull.eval(o2).flatMap(o => Pull.output1(o))`

**Comment by mpilquist** [comment]
Thanks!

**Pull request merged**